web-services/query/src/main/java/datawave/webservice/query/runner/QueryExecutorBean.java

```diff
@@ -3212,79 +3212,80 @@ public void write(OutputStream out) throws IOException, WebApplicationException
-                        
-                        long serializationStart = System.nanoTime();
-                        switch (serializationType) {
-                            case XML:
-                                xmlSerializer.marshal(page, countingStream);
-                                break;
-                            case JSON:
-                                // First page!
-                                if (!sentResults) {
-                                    jsonGenerator.writeStartObject();
-                                    jsonGenerator.writeArrayFieldStart("Pages");
-                                    jsonGenerator.flush();
-                                } else {
-                                    // Delimiter for subsequent pages...
-                                    countingStream.write(',');
-                                }
-                                jsonSerializer.writeValue(countingStream, page);
-                                break;
-                            case PB:
-                                @SuppressWarnings("unchecked")
-                                Message<Object> pb = (Message<Object>) page;
-                                Schema<Object> pbSchema = pb.cachedSchema();
-                                ProtobufIOUtil.writeTo(countingStream, page, pbSchema, buffer);
-                                buffer.clear();
-                                break;
-                            case YAML:
-                                @SuppressWarnings("unchecked")
-                                Message<Object> yaml = (Message<Object>) page;
-                                Schema<Object> yamlSchema = yaml.cachedSchema();
-                                YamlIOUtil.writeTo(countingStream, page, yamlSchema, buffer);
-                                buffer.clear();
-                                break;
-                        }
-                        countingStream.flush();
-                        long serializationTime = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - serializationStart);
-                        pm.setSerializationTime(serializationTime);
-                        long pageCallTime = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - callStart);
-                        pm.setCallTime(pageCallTime);
-                        pm.setBytesWritten(countingStream.getCount());
-                        sentResults = true;
-                    } catch (Exception e) {
-                        if (e instanceof NoResultsException || e.getCause() instanceof NoResultsException) {
-                            // No more results, break out of loop
-                            done = true;
-                            break; // probably redundant
-                        } else {
-                            throw e;
+                try (JsonGenerator jsonGenerator = jsonSerializer.getFactory().createGenerator(out, JsonEncoding.UTF8)) {
```

Comment:
Are we sure that closing the json generator won't close the underlying output stream? We definitely can't close out, since that's provided by Wildfly.

Reply:
It does not. The close method on JsonGenerator implementations will only close if the resource is managed or if JsonGenerator.Feature.AUTO_CLOSE_TARGET is true.. The createGenereator method sets the managed resource flag to false and we set AUTO_CLOSE_TARGET explicitly above.